engine/fasthttp/request.go

```diff
@@ -85,7 +85,7 @@ func (r *Request) Method() string {
 
 // SetMethod implements `engine.Request#SetMethod` function.
 func (r *Request) SetMethod(method string) {
-	r.Request.Header.SetMethod(method)
+	r.Request.Header.SetMethodBytes([]byte(method))
```

Comment:
No unfortunately, it appends it, so if we have "GET" and set it to "POST" it wold be "GETPOST"

https://github.com/valyala/fasthttp/blob/b8b31526d7a61e3d43e9ae02aeed8a4ea6adf775/header.go#L464